src/applications/facility-locator/components/AppointmentInfo.jsx

```diff
@@ -71,72 +66,69 @@ export default class AppointmentInfo extends Component {
       return null;
     };
 
-    const renderSpecialtyTimes = (existing = false) => {
-      const specialtyKeys = healthAccessAttrs && Object.keys(healthAccessAttrs);
-      pull(specialtyKeys, 'primaryCare', 'effectiveDate');
-      specialtyKeys.sort();
-
-      if (specialtyKeys && specialtyKeys.length === 0) {
-        return null;
-      }
-
-      const firstThree = specialtyKeys.slice(0, 3);
-      const lastToEnd = specialtyKeys.slice(3);
-      let showHideKey;
-
-      if (existing) {
-        showHideKey = 'newPatientTimesExpanded';
+    const renderPrimaryCare = (accessAttrs, category) => {
+      if (Array.isArray(accessAttrs)) {
+        // V1
+        const value = accessAttrs.find(
+          k => k.service === 'PrimaryCare' && k[category],
+        );
+        return renderStat('Primary Care', value[category]);
       } else {
-        showHideKey = 'existingPatientTimesExpanded';
+        // V0 legacy structure TODO: remove when fully migrated to V1
+        const value = get(accessAttrs, ['primaryCare', category]);
+        if (value) {
+          return renderStat('Primary Care', value);
+        }
       }
+      return null;
+    };
 
-      const onClick = () => {
-        recordEvent({ event: 'fl-show-waittimes' });
-        this.setState({
-          [showHideKey]: !this.state[showHideKey],
-        });
-      };
+    const renderSpecialtyTimes = (existing = false) => {
+      if (Array.isArray(healthAccessAttrs)) {
+        // V1
+        const healthAccessSpecialty = healthAccessAttrs.filter(
+          acc => acc.service !== 'PrimaryCare',
+        );
+        return (
+          <li key="specialty-care">
+            Specialty care:
+            <ul className="vads-u-margin-top--1">
+              {healthAccessSpecialty.map(k =>
+                renderStat(
+                  k.service.replace(/([A-Z])/g, ' $1').trim(),
```

Comment:
This regex is being used in multiple places. Let's move it into `renderStat`, and add a unit test for `renderStat` to make sure it does what we expect.